app/models/visitors_step.rb

```diff
@@ -79,9 +84,41 @@ def validate_email
     end
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
```

Comment:
I agree with rubocop, I'd find the method more readable if it was split up, for example one part doing the api call and checking the result, and another adding the errors to the objects if it is invalid.